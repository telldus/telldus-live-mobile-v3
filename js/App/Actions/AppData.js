/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

// @flow

'use strict';

import type { ThunkAction } from './Types';
import { actions } from 'live-shared-data';
const {Devices: {getDevices}} = actions;
const {Sensors: {getSensors}} = actions;
const {Jobs: {getJobs}} = actions;
const {Websockets: {authenticateSession, connectToGateways}} = actions;

import { widgetAndroidConfigureSessionData } from './Widget';

function getAppData(): ThunkAction {
	return (dispatch: Function, getState: Object) => {
		dispatch(authenticateSession()).then((sessionId: string) => {
			if (sessionId) {
				dispatch(widgetAndroidConfigureSessionData(sessionId));
			}
		});
		dispatch(connectToGateways());

		dispatch(getDevices());
		dispatch(getSensors());
		dispatch(getJobs());
	};
}

module.exports = {
	getAppData,
};
