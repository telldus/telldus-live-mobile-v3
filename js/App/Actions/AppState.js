/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @providesModule Actions_AppState
 */

// @flow

'use strict';

import type { Action, ThunkAction } from './Types';

import { AppState } from 'react-native';
import Orientation from 'react-native-orientation';
import { AccessibilityInfo } from 'react-native';

module.exports = {
	appStart: (): Action => ({
		type: 'APP_START',
	}),
	appState: (): ThunkAction => dispatch => {
		AppState.addEventListener('change', appState => {
			if (appState === 'active') {
				return dispatch({
					type: 'APP_FOREGROUND',
				});
			}
			if (appState === 'background') {
				return dispatch({
					type: 'APP_BACKGROUND',
				});
			}
		});
	},
	appOrientation: (initialOrientation: string): ThunkAction => dispatch => {
		Orientation.addOrientationListener((orientation: string) => {
			return dispatch({
				type: 'APP_ORIENTATION',
				value: orientation,
			});
		});
		return dispatch({
			type: 'APP_ORIENTATION',
			value: initialOrientation,
		});
	},
	setAppLayout: (layout: Object): ThunkAction => dispatch => {
		return dispatch({
			type: 'APP_LAYOUT',
			payload: layout,
		});
	},
	setAccessibilityInfo: (status: boolean): ThunkAction => dispatch => {
		return dispatch({
			type: 'ACCESSIBILITY_INFO',
			payload: status,
		});
	},
	setAccessibilityListener: (callback: Function): ThunkAction => dispatch => {
		AccessibilityInfo.addEventListener(
			'change',
			(isEnabled: boolean) => {
				return dispatch(callback(isEnabled));
			}
		);
	},
	setAppVersion: (appVersion: string): ThunkAction => dispatch => {
		return dispatch({
			type: 'SET_APP_VERSION',
			payload: appVersion,
		});
	},
};
