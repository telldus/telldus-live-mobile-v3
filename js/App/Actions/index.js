/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

// @flow

'use strict';

import GatewayActions from './Gateways';
import LoginActions from './Login';
import NavigationActions from './Navigation';
import SensorActions from './Sensors';
import DeviceActions from './Devices';
import LiveApiActions from './LiveApi';
import DashboardActions from './Dashboard';
import JobActions from './Jobs';
import AppStateActions from './AppState';
import Schedule from './Schedule';
import AppDataActions from './AppData';
import ModalActions from './Modal';
import WebSocketActions from './Websockets';
import AppActions from './App';
import ScheduleActions from './Schedule';
import UserActions from './User';
import LocalTest from './LocalTest';
import LocalControl from './LocalControl';
import Widget from './Widget';
import AddDevice from './AddDevice';
import * as Fences from './Fences';
import * as Events from './Events';
import * as GeoFence from './GeoFence';
import * as Firebase from './Firebase';
import * as Auth from './Auth';
import * as ThirdParties from './ThirdParties';
import * as WebsocketExtras from './WebsocketExtras';
import * as Event from './Event';

module.exports = {
	...GatewayActions,
	...LoginActions,
	...NavigationActions,
	...SensorActions,
	...DeviceActions,
	...LiveApiActions,
	...DashboardActions,
	...JobActions,
	...AppStateActions,
	...Schedule,
	...AppDataActions,
	...ModalActions,
	...WebSocketActions,
	...AppActions,
	...ScheduleActions,
	...UserActions,
	...LocalTest,
	...LocalControl,
	...Widget,
	...AddDevice,
	...Fences,
	...Events,
	...GeoFence,
	...Firebase,
	...Auth,
	...ThirdParties,
	...WebsocketExtras,
	...Event,
};
