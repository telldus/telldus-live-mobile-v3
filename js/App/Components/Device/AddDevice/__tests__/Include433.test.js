/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope this it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 *
 */


import React from 'react';
import {act} from 'react-test-renderer';
import { connect } from 'react-redux';

import {
	rendererWithIntlAndReduxProviders,
	setAppLayoutInStore,
	NAVIGATION_PROP,
	DUMMY_DEVICE_433,
	DEVICE_MANU_INFO_433,
	DUMMY_CLIENT,
	setDeviceListInStore,
	setGatewaysListInStore,
	setDeviceInfoInStore,
	getStore,
} from '../../../../../Utils/jestUtils';
import Include433 from '../Include433';

import * as Actions from '../../../../Actions';
import {
	AddDeviceContainer as UnConnectedAddDeviceContainer,
	mapDispatchToProps,
	mapStateToProps,
} from '../AddDeviceContainer';

const store = getStore();

const intl = {
	formatMessage: jest.fn(() => ''),
};
const appLayout = {
	height: 100,
	width: 100,
};
const onDidMount = jest.fn();
const deviceName = 'Device name';
const navigation = {
	...NAVIGATION_PROP,
};
const route = {
	params: {
		gateway: {
			id: DUMMY_DEVICE_433.client,
		},
		deviceInfo: {
			...DEVICE_MANU_INFO_433,
		},
		deviceName,
	},
};
const currentScreen = 'Include433';
const ScreenName = currentScreen;

let actions = {
	initiateAdd433MHz: jest.fn((id, deviceInfo, formatMessage) => {
		return Promise.resolve(() => {});
	}),
};

const initiate = () => {
	setAppLayoutInStore();
	setDeviceListInStore();
	setGatewaysListInStore();
	setDeviceInfoInStore();
	store.dispatch({
		type: 'RECEIVED_GATEWAY_WEBSOCKET_ADDRESS',
		gatewayId: DUMMY_CLIENT.id,
		payload: {
			address: 123,
			instance: 123,
			port: 123,
		},
	});
};

const customMapStateToProps = (_store, _ownProps) => {
	return {
		...mapStateToProps(_store, _ownProps),
		currentScreen,
	};
};

jest.useFakeTimers();

describe('<Include433 />', () => {

	beforeAll(() => {
		actions.initiateAdd433MHz.mockReset();
		actions = {
			...actions,
			initiateAdd433MHz: jest.fn((id, deviceInfo, formatMessage) => {
				return store.dispatch(Actions.initiateAdd433MHz(id, deviceInfo, formatMessage));
			}),
		};
	});

	it('Should initiateAdd433MHz and show activity indicator', () => {

		let component;
		act(() => {
			initiate();
			component = rendererWithIntlAndReduxProviders(
				<Include433
					navigation={navigation}
					route={route}
					intl={intl}
					appLayout={appLayout}
					onDidMount={onDidMount}
					actions={actions}/>
			);
		});

		const args = [
			DUMMY_DEVICE_433.client,
			{
				...DEVICE_MANU_INFO_433,
				deviceName,
			},
			intl.formatMessage,
		];

		const root = component.root;
		const childInclude433 = root.findByType(Include433);
		const childInclude433Instance = childInclude433.instance;

		expect(childInclude433Instance.deleteSocketAndTimer).toBeInstanceOf(Function);

		expect(actions.initiateAdd433MHz).toBeCalledTimes(1);
		expect(actions.initiateAdd433MHz).toBeCalledWith(...args);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe('<Include433 /> with container component', () => {
	const deleteSocketAndTimer = jest.fn(() => {});

	const initiateAdd433MHz = jest.fn(() => {
		return deleteSocketAndTimer;
	});

	const customMapDispatchToProps = (dispatch) => {
		const {
			actions: actionsProp,
			...others
		} = mapDispatchToProps(dispatch);
		return {
			...others,
			actions: {
				...actionsProp,
				initiateAdd433MHz,
			},
		};
	};

	const ConnectedAddDeviceContainer = connect(customMapStateToProps, customMapDispatchToProps)(UnConnectedAddDeviceContainer);

	afterAll(() => {
		initiateAdd433MHz.mockReset();
		deleteSocketAndTimer.mockReset();
	});

	it('Should initiateAdd433MHz and show activity indicator with container', () => {

		let component;
		act(() => {
			initiate();

			store.dispatch(Actions.addDevice433Initiate());

			const screenProps = {
				appLayout,
				intl,
			};
			component = rendererWithIntlAndReduxProviders(
				<ConnectedAddDeviceContainer
					screenProps={screenProps}
					navigation={navigation}
					currentScreen={currentScreen}
					ScreenName={ScreenName}
					route={route}
					children={
						<Include433/>
					}
				/>
			);
		});

		const root = component.root;
		const childInclude433 = root.findByType(Include433);
		const childInclude433Instance = childInclude433.instance;

		expect(childInclude433Instance.deleteSocketAndTimer).toBeInstanceOf(jest.fn().constructor);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe('<Include433 /> on success with container component', () => {

	const deleteSocketAndTimer = jest.fn(() => {}).mockName('deleteSocketAndTimer');

	const dummyDeviceId = DUMMY_DEVICE_433.id;
	const initiateAdd433MHz = jest.fn(() => {
		return deleteSocketAndTimer;
	}).mockName('initiateAdd433MHz');

	const customMapDispatchToProps2 = (dispatch) => {
		const {
			actions: actionsProp,
			...others
		} = mapDispatchToProps(dispatch);
		return {
			...others,
			actions: {
				...actionsProp,
				initiateAdd433MHz,
			},
		};
	};

	const ConnectedAddDeviceContainer = connect(customMapStateToProps, customMapDispatchToProps2)(UnConnectedAddDeviceContainer);

	afterAll(() => {
		initiateAdd433MHz.mockReset();
		deleteSocketAndTimer.mockReset();
	});

	it('Should initiateAdd433MHz and show success screen with container, also call deleteSocketAndTimer when unmount', () => {

		let component;
		act(() => {
			initiate();

			// Mocking the add device success here
			store.dispatch(Actions.addDevice433Initiate());
			store.dispatch(Actions.addDevice433Success(dummyDeviceId));

			const screenProps = {
				appLayout,
				intl,
			};
			component = rendererWithIntlAndReduxProviders(
				<ConnectedAddDeviceContainer
					screenProps={screenProps}
					currentScreen={currentScreen}
					ScreenName={ScreenName}
					navigation={navigation}
					route={route}
					children={
						<Include433/>
					}
				/>
			);
		});

		const root = component.root;
		const childInclude433 = root.findByType(Include433);
		const childInclude433Instance = childInclude433.instance;

		expect(childInclude433Instance.deleteSocketAndTimer).toBeInstanceOf(jest.fn().constructor);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		component.unmount();

		expect(deleteSocketAndTimer).toBeCalledTimes(1);
	});
});

describe('<Include433 /> on error with container component', () => {

	const deleteSocketAndTimer = jest.fn(() => {}).mockName('deleteSocketAndTimer');

	const initiateAdd433MHz = jest.fn(() => {
		return deleteSocketAndTimer;
	}).mockName('initiateAdd433MHz');

	const customMapDispatchToProps = (dispatch) => {
		const {
			actions: actionsProp,
			...others
		} = mapDispatchToProps(dispatch);
		return {
			...others,
			actions: {
				...actionsProp,
				initiateAdd433MHz,
			},
		};
	};

	const ConnectedAddDeviceContainer = connect(customMapStateToProps, customMapDispatchToProps)(UnConnectedAddDeviceContainer);

	afterAll(() => {
		initiateAdd433MHz.mockReset();
		deleteSocketAndTimer.mockReset();
	});

	it('Should initiateAdd433MHz and show error screen with container, also call deleteSocketAndTimer when unmount', () => {

		let component;
		act(() => {
			initiate();

			// Mocking the add device fail here
			store.dispatch(Actions.addDevice433Initiate());
			store.dispatch(Actions.addDevice433Failed('Error when add device 433MHz failed'));

			const screenProps = {
				appLayout,
				intl,
			};
			component = rendererWithIntlAndReduxProviders(
				<ConnectedAddDeviceContainer
					screenProps={screenProps}
					currentScreen={currentScreen}
					ScreenName={ScreenName}
					navigation={navigation}
					route={route}
					children={
						<Include433/>
					}
				/>
			);
		});

		const root = component.root;
		const childInclude433 = root.findByType(Include433);
		const childInclude433Instance = childInclude433.instance;

		expect(childInclude433Instance.deleteSocketAndTimer).toBeInstanceOf(jest.fn().constructor);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		component.unmount();

		expect(deleteSocketAndTimer).toBeCalledTimes(1);
	});
});
