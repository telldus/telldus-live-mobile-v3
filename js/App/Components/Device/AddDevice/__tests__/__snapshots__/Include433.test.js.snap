// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Include433 /> Should initiateAdd433MHz and show activity indicator 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`<Include433 /> on error with container component Should initiateAdd433MHz and show error screen with container, also call deleteSocketAndTimer when unmount 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeeee",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#1b365d",
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "paddingHorizontal": 15,
              "paddingTop": 0,
              "position": "relative",
            },
            Object {
              "height": 106.72,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "telldus_logo",
              }
            }
            style={
              Array [
                Object {
                  "height": 35.00000025,
                  "resizeMode": "contain",
                  "width": 230.49999975,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "height": 15.5,
              "overflow": "hidden",
              "width": "100%",
            }
          }
        >
          <View
            fill="#e26901"
            height={313}
            style={
              Object {
                "marginTop": -2,
              }
            }
            width={750}
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "backgroundColor": "transparent",
                    "flex": 0,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginRight": 12.4,
                    "width": 80,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "elevation": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 1.38885,
            "padding": 2.7777,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.23,
            "shadowRadius": 2,
          },
          Array [
            Object {
              "flex": 0,
              "marginBottom": 1.38885,
              "marginHorizontal": 2.7777,
            },
            Object {
              "marginVertical": 2.7777,
            },
          ],
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#1b365d",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          importantForAccessibility="yes"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#e26901",
                "fontFamily": "telldusicons",
                "fontSize": 14.000000000000002,
              },
              Object {
                "color": "#D32F2E",
              },
            ]
          }
        >
          info
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#555555",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 4.2,
                "marginLeft": 2.7777,
              },
              Object {
                "fontSize": 3.5000000000000004,
              },
            ]
          }
        >
          Error when add device 433MHz failed
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Include433 /> on success with container component Should initiateAdd433MHz and show success screen with container, also call deleteSocketAndTimer when unmount 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeeee",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#1b365d",
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "paddingHorizontal": 15,
              "paddingTop": 0,
              "position": "relative",
            },
            Object {
              "height": 106.72,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "telldus_logo",
              }
            }
            style={
              Array [
                Object {
                  "height": 35.00000025,
                  "resizeMode": "contain",
                  "width": 230.49999975,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "height": 15.5,
              "overflow": "hidden",
              "width": "100%",
            }
          }
        >
          <View
            fill="#e26901"
            height={313}
            style={
              Object {
                "marginTop": -2,
              }
            }
            width={750}
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "backgroundColor": "transparent",
                    "flex": 0,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginRight": 12.4,
                    "width": 80,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flexGrow": 1,
            "paddingVertical": 2.7777,
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "elevation": 2,
                  "marginBottom": 10.416375,
                  "marginHorizontal": 20.83275,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.23,
                  "shadowRadius": 2,
                  "width": 708.3345,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "stretch",
                    "backgroundColor": "transparent",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "paddingVertical": 72,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderRightColor": "transparent",
                      "borderRightWidth": 37.5,
                      "borderStyle": "solid",
                      "borderTopColor": "#1b365d",
                      "borderTopLeftRadius": 2,
                      "borderTopWidth": 120,
                      "height": 0,
                      "left": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": 150,
                    }
                  }
                />
                <Text
                  allowFontScaling={false}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#fff",
                      "fontSize": 37.5,
                      "left": 45,
                      "position": "absolute",
                      "top": 7.5,
                    }
                  }
                >
                  1.
                </Text>
                <View
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "height": 0,
                      "width": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                    "paddingHorizontal": 20.83275,
                    "paddingVertical": 41.6655,
                    "width": 555,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#8e8e93",
                      "flexWrap": "wrap",
                      "fontSize": 26.250000000000004,
                    }
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#8e8e93",
                    "fontSize": 3,
                    "marginBottom": 4,
                  }
                }
              >
                (0% )
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#cccccc",
                      "borderColor": "transparent",
                      "borderRadius": 4,
                      "borderWidth": 0,
                      "justifyContent": "center",
                      "overflow": "hidden",
                      "width": 88.8892,
                    },
                    Object {
                      "marginBottom": 2.7777,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#e26901",
                      "height": 4,
                      "transform": Array [
                        Object {
                          "translateX": -3.552713678800501e-15,
                        },
                        Object {
                          "translateX": -44.44015554,
                        },
                        Object {
                          "scaleX": 0.0001,
                        },
                      ],
                      "width": 88.8892,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "elevation": 2,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 1.38885,
                  "padding": 2.7777,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.23,
                  "shadowRadius": 2,
                },
                Object {
                  "flex": 0,
                  "marginBottom": 1.38885,
                  "marginHorizontal": 2.7777,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#1b365d",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                importantForAccessibility="yes"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#e26901",
                      "fontFamily": "telldusicons",
                      "fontSize": 14.000000000000002,
                    },
                    undefined,
                  ]
                }
              >
                info
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#555555",
                      "flex": 1,
                      "flexWrap": "wrap",
                      "fontSize": 4.2,
                      "marginLeft": 2.7777,
                    },
                    Object {
                      "fontSize": 3.5000000000000004,
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;

exports[`<Include433 /> with container component Should initiateAdd433MHz and show activity indicator with container 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeeee",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#1b365d",
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "paddingHorizontal": 15,
              "paddingTop": 0,
              "position": "relative",
            },
            Object {
              "height": 106.72,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "telldus_logo",
              }
            }
            style={
              Array [
                Object {
                  "height": 35.00000025,
                  "resizeMode": "contain",
                  "width": 230.49999975,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "height": 15.5,
              "overflow": "hidden",
              "width": "100%",
            }
          }
        >
          <View
            fill="#e26901"
            height={313}
            style={
              Object {
                "marginTop": -2,
              }
            }
            width={750}
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "backgroundColor": "transparent",
                    "flex": 0,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginRight": 12.4,
                    "width": 80,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;
